presto-native-execution/src/test/java/com/facebook/presto/nativeworker/SymlinkManifestGeneratorUtils.java

```diff
@@ -0,0 +1,151 @@
+     * to a symlink manifest file named "symlink_manifest" inside the specified symlinkManifestDir.
+     *
+     * @param hiveTableLocation the root directory of the Hive table files
+     * @param symlinkManifestDir the directory where the symlink manifest should be created
+     * @throws IOException if an I/O error occurs while creating the directories or writing the manifest
+     */
+    public static void createSymlinkManifest(Path hiveTableLocation, Path symlinkManifestDir) throws IOException
+    {
+        if (Files.notExists(symlinkManifestDir.getParent())) {
+            Files.createDirectory(symlinkManifestDir.getParent());
+        }
+
+        if (Files.notExists(symlinkManifestDir)) {
+            Files.createDirectory(symlinkManifestDir);
+        }
+
+        Path manifestFilePath = symlinkManifestDir.resolve("symlink_manifest");
+        try (BufferedWriter writer = Files.newBufferedWriter(manifestFilePath, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING)) {
+            List<String> fileList = new ArrayList<>();
+            collectDataFiles(hiveTableLocation, fileList);
+
+            for (String filePath : fileList) {
+                writer.write(filePath);
+                writer.write(System.lineSeparator());
+            }
+        }
+    }
+
+    /**
+     * Recursively collects all non-hidden file paths in the source directory.
+     * If a directory is encountered, the method is called recursively to collect files in subdirectories.
+     *
+     * @param sourceDir the directory to start collecting files from
+     * @param fileList the list where the collected file paths are stored
+     * @throws IOException if an I/O error occurs while accessing the directory or files
+     */
+    private static void collectDataFiles(Path sourceDir, List<String> fileList) throws IOException
+    {
+        try (DirectoryStream<Path> stream = Files.newDirectoryStream(sourceDir)) {
+            for (Path entry : stream) {
+                if (!Files.isHidden(entry)) {
+                    if (Files.isDirectory(entry)) {
+                        collectDataFiles(entry, fileList);
+                    }
+                    else {
+                        fileList.add(entry.toString());
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Cleans up (deletes) all files and subdirectories inside the specified directory.
+     * This method walks through the directory tree and deletes files in reverse order to ensure
+     * that directories are deleted after their contents.
+     *
+     * @param directory the root directory to clean up
+     * @throws IOException if an I/O error occurs during file deletion
+     */
+
+    public static void cleanupSymlinkData(Path directory) throws IOException
+    {
+        MoreFiles.deleteRecursively(directory, RecursiveDeleteOption.ALLOW_INSECURE);
+    }
+
+    /**
+     * Returns the Hive type string corresponding to a given TPCH type.
```

Comment:
Don't follow this... Is integer the only exception for type strings ?

Reply:
Serde constant for integer is `int` from hive library https://github.com/apache/hive/blob/master/serde/src/gen/thrift/gen-javabean/org/apache/hadoop/hive/serde/serdeConstants.java#L58 So I am returning that from here.